### PR TITLE
examples/gcoap-rust: Expose more functionality

### PR DESCRIPTION
**Pull request by chrysn**
### Contribution description

The riot-coap-handler-demos has a few ready-made components that were not shown off in examples/rust-gcoap before. I think they are useful to have, as they point users to more example code and to things they can explore with safe Rust bindings. The added resources don't make the example's source code more complex (effectively they're all just adding a line), but obviously they increase code size.

Concretely, this exposes any board LEDs as individual resources (PUT CBOR values false or true, or POST (see maturity)), and also exposes /netif which roughly contains data from `ifconfig` (only MAC address and IP addresses so far).

Two non-src/lib.rs changes are an added note in the README on security (none is there, thus we only expose harmless things), updating various riot-wrappers related dependencies (so for example LEDs are properly wrapped in the first place), and a stack size increase.

The CoAP thread's stack size gets increased from 8k to 16k. This fixes an actual bug, where on the particle-xenon the stack usage of the /ps/ handler (that was already active) was around 11k. I've done some digging into it (with terrible methodology, a mix of interactive gdb, attempts to read the stack usage from a Python script in gdb, and eventually resorting to calling ps() over and over in various places because that's faster than breakpoints), and what I can tell so far is that it's not so much related to serialization (I've tried switching CBOR libraries but the effect was present but miniomal) but that a lot of stack is used already in request parsing. Clearly something to investigate further, but for now, easy to fix.

### Testing procedure

* Run the examples/rust-gcoap program and note the IP address it shows
* `pipx run --spec 'aiocoap[prettyprint]' aiocoap-client coap://[fe80::the-address-shown%relevant-netif]/.well-known/core` (or just aiocoap-client if you have aiocoap installed) and from there interact with the other resources advertised

The prettyprinting will make sure you see the returned CBOR in a reasonable form; very recent (I think unpublished) versions even give you readable IP addesses such as this:

```
./aiocoap-client coap://'[fe80::a41b:b9ff:fe1f:0894%enxa61bb91f1494]'/netif
# CBOR message shown in Diagnostic Notation
[_ [6, h'a61bb91f0894', [54([ip'fe80::a41b:b9ff:fe1f:894', null, 6])]]]
```

(which would be a lot more readable even if tag 54's text notation had the capability of expressing zone identifiers).

### Maturity

<del>The LED resources are a bit weird so far in that toggling them requires sending a null value rather than an empty payload; I'd like to fix that before merging.</del>

### CODE REVIEW

**Comment by chrysn** [comment]
LEDs can now be toggled with an empty POST, so the maturity concern there is done.

(Still doesn't say that all resource implementations are particularly high-quality -- but after all, this *is* a demo example).

**Comment by chrysn** [comment]
Before passing this will take another tour through generate-Makefile.ci, but I'd rather do that only after an initial review, because running that takes hours.

**Review by mcr — APPROVED** [review]

File: examples/rust-gcoap/Makefile [hunk 1]
```diff
@@ -34,10 +34,9 @@ DEVELHELP ?= 1
 # Change this to 0 show compiler invocation lines by default:
 QUIET ?= 1
 
-# Add 8k extra stack: The Rust examples take more of it than gcoap expects,
-# presumably because the example use the standard library's sting formatting
-# instead of one of the more optimized formatters.
-CFLAGS += -DGCOAP_STACK_SIZE='(THREAD_STACKSIZE_DEFAULT+DEBUG_EXTRA_STACKSIZE+sizeof(coap_pkt_t)+8192)'
+# Add 16k extra stack: The Rust examples take more of it than gcoap expects,
```
Comment (Teufelchen1) [inline examples/rust-gcoap/Makefile:37]: could you try it with e.g. 12k, to see if it still fits on the sam* devices? I feel those might be common enough to justify breaking the beauty of 2^x :)

**Comment by chrysn** [comment]
I've taken the liberty to add a ping example.

@teufelchen, you may want to have fun with that from your console! Both [ping](https://doc.riot-os.org/rustdoc/latest/riot_coap_handler_demos/ping/index.html) and [pinged](https://doc.riot-os.org/rustdoc/latest/riot_coap_handler_demos/ping_passive/index.html) are discoverable through CoAP if= properties, and while I make no promises on the textual format, the (anyway more-fun-because-it-has-more-data) CBOR that comes even with some documentation.

File: examples/rust-gcoap/Makefile (hunk 1 above)
Comment (chrysn) [reply examples/rust-gcoap/Makefile:39]: Indeed it fits, and shaved 1k of the main thread too. generate-Makefile.ci in progress, should suffice for them!

**Comment by chrysn** [comment]
A new addition patches the try-lock crate to use its PR'd https://github.com/seanmonstar/try-lock/pull/11 instead of its released version: as so often, armv6's lacking atomics bite and need a workaround.

File: examples/rust-gcoap/Makefile (hunk 1 above)
Comment (Teufelchen1) [reply examples/rust-gcoap/Makefile:39]:
```suggestion
# Add 12k extra stack: The Rust examples take more of it than gcoap expects,
```

**Review by Teufelchen1 — APPROVED** [review]
Works on my nrf52840dk :)

**Comment by chrysn** [comment]
Thanks for the review, squashed.

**Pull request merged**